class-plugin-autoupdate-filter.php

```diff
@@ -12,9 +12,9 @@
 class Plugin_Autoupdate_Filter {
 
 	/**
-	 * Plugin_Autoupdate_Filter constructor.
+	 * Initialize WordPress hooks
 	 */
-	public function __construct() {
+	public function init() {
```

Comment:
I believe the is the best way to do it. Could we accomplish something similar with a Singleton Pattern? Yes, probably, but that would also probably be using the pattern for something other than the way it was originally intended. Let's tease this out a little bit.

The most common example for a Singleton Pattern is a database access class, where you only want one instance of that class to exist, because there are potential bad side-effects of having two or more of those objects interacting with the database. 

In our case, what would be the bad side-effects if two instances of this class were instantiated? Not much, really, just some filters being added twice. This might happen in a case of two versions of the same plugin running on the same site. Nothing catastrophic, IMO.

In fact, if we wanted to look more closely at how we are using classes in the plugin, we probably don't even need to use classes. We are using classes to prevent collisions with other code; to separate our code out from any other functions in other custom code or plugins that might use a function of the same name. If we wanted to, we _could_ use namespaces to prevent collision with other code. We'll leave it for now, but something to think about when creating little plugins like this in the future.

Reply:
Side note: I'm fully aware that I'm the one who originally crafted this plugin using classes, and that I'm bringing my own design choice up for debate 😄

Reply:
> Can you please figure out the best way to not include the .DS_Store file from your computer in these changes?

Oops! Thank you for catching that. Removed :) 

> If we wanted to, we could use namespaces to prevent collision with other code. We'll leave it for now, but something to think about when creating little plugins like this in the future.

That makes sense. In this case, the class really is just for namespace reasons. That was another question I had while looking at this. Why classes and not namespaces? I know there's been an effort on the Woo side to refactor and switch over to namespaces. 

Related - I couldn't think of any reason someone might want to instantiate a second instance of this class. However, theoretically if there were, that would be an argument for class over namespace. Could there be some benefit from the extensibility side to handling this as a class? Because it makes it more extensible for cases we can't imagine right now?

Reply:
> Could there be some benefit from the extensibility side to handling this as a class?

Right! That's one great benefit of classes. An example where I've used that extensibility fairly recently is by [extending an Action Scheduler class](https://github.com/a8cteam51/safety-net/blob/trunk/includes/classes/class-actionscheduler-custom-dbstore.php) in order to pause renewal actions. Without the ability to extend that class, our job would've been significantly more difficult.

But, like you said, in this case, with this tiny little plugin which is being distributed mostly only in the context of our own sites we manage, I can't imagine a lot of people needing to extend our classes. In most of these types of cases, we could probably just get away with namespacing it.